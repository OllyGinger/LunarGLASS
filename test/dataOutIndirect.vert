#version 130

attribute vec4 color;
varying vec4 colorOut[6];

void main()
{
    for (int i = 1; i < 5; ++i)
        colorOut[i] = color;

    gl_Position = colorOut[2];
}
