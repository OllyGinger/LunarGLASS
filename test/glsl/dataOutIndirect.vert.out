
***Unsupported functionality: unsupported array dereference
compiling dataOutIndirect.vert...
attribute vec4 color ; 
varying vec4 colorOut [ 6 ] ; 
void  main (){
for( int i = 1 ; ; i < 5 ; ++ i ) colorOut [ i ] = color ; gl_Position = colorOut [ 2 ] ; }



(
(declare (out ) (array vec4 7) colorOut)
(declare (in ) vec4 color)
(declare (out ) vec4 gl_Position)
(function main
  (signature void
    (parameters
    )
    (
      (assign (constant bool (1)) (xyzw) (array_ref (var_ref colorOut) (constant int (1)) )  (var_ref color) ) 
      (assign (constant bool (1)) (xyzw) (array_ref (var_ref colorOut) (constant int (2)) )  (var_ref color) ) 
      (assign (constant bool (1)) (xyzw) (array_ref (var_ref colorOut) (constant int (3)) )  (var_ref color) ) 
      (assign (constant bool (1)) (xyzw) (array_ref (var_ref colorOut) (constant int (4)) )  (var_ref color) ) 
      (assign (constant bool (1)) (xyzw) (var_ref gl_Position)  (array_ref (var_ref colorOut) (constant int (2)) ) ) 
    ))

)


)